staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -511,6 +511,13 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 		return newErrWatcher(err), nil
 	}
 
+	// FIXME: Change the condition to ensure that we're not allocating memory for initEvents
+	//   without any reasonable boundary.
+	if false {
+		// TODO: See above - can 429 be returned as error - or the above argument holds here too?
+		return nil, errors.NewTooManyRequests("<TODO>", 1)
```

Comment:
does any API server path today return a 429 directly to a watcher (vs in a status object in an ERROR watch event)? how do old clients (informer, watchtools, etc) handle that?

Reply:
Non that I was able to find. (basically we're not rate-limitting watches at all).